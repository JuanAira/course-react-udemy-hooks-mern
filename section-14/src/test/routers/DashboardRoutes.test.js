import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import { DashboardRoutes } from '../../routers/DashboardRoutes';

describe('Tests in <DashboardRoutes />', () => {
    const contextValue = {
        user: {
            logged: true,
            name: 'Juanito'
        }
    };

    test('should show to match snapshot', () => {

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/']}>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Juanito');
        expect(wrapper.find('h1').text().trim()).toBe('MarvelScreen');
    });

    test('should show correct DC', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/dc']}>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe('DCScreen');
    });
});
