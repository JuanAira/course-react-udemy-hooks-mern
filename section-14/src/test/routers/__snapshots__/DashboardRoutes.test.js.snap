// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <DashboardRoutes /> should show correct DC 1`] = `ReactWrapper {}`;

exports[`Tests in <DashboardRoutes /> should show to match snapshot 1`] = `ReactWrapper {}`;
